### PR TITLE
Add propagation helpers for the "traceparent:" header.

### PR DESCRIPTION
**Pull request by g-easy**

### CODE REVIEW

File: opencensus/trace/BUILD [hunk 1]
```diff
@@ -135,6 +136,23 @@ cc_library(
     ],
 )
 
+cc_library(
+    name = "traceparent",
```
Comment (bogdandrutu) [inline opencensus/trace/BUILD]: I would call this trace context to be consistent with all the other languages

File: opencensus/trace/internal/traceparent_test.cc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  INVALID("00-404142434445464748494A4B4C4D4E4F-6X6Y636465666768-01")
+      << "span_id not hex.";
+  INVALID("00-404142434445464748494A4B4C4D4E4F-6162636465666768-XY")
+      << "options not hex.";
+  INVALID("00-00000000000000000000000000000000-6162636465666768-01")
+      << "trace_id can't be zero.";
+  INVALID("00-404142434445464748494A4B4C4D4E4F-0000000000000000-01")
+      << "span_id can't be zero.";
+  INVALID("00_404142434445464748494A4B4C4D4E4F-6162636465666768-01")
+      << "invalid delimiter.";
```
[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/trace/internal/traceparent_test.cc]: Done.

File: opencensus/trace/internal/traceparent_test.cc [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  INVALID("00-404142434445464748494A4B4C4D4E4F-6162636465666768-0")
+      << "too short.";
+  INVALID("00-404142434445464748494A4B4C4D4E4F-6162636465666768-011")
+      << "too long.";
+  INVALID("00-4X4Y42434445464748494A4B4C4D4E4F-6162636465666768-01")
+      << "trace_id not hex.";
+  INVALID("00-404142434445464748494A4B4C4D4E4F-6X6Y636465666768-01")
+      << "span_id not hex.";
+  INVALID("00-404142434445464748494A4B4C4D4E4F-6162636465666768-XY")
+      << "options not hex.";
```
[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/trace/internal/traceparent_test.cc]: Done. (version was the only bit missing)

File: opencensus/trace/BUILD (hunk 1 above)
Comment (g-easy) [reply opencensus/trace/BUILD]: Done. Should I also rename the functions from TraceParent? (I find it a bit weird that the header doesn't match the other names but consistency is important)

**Comment by g-easy** [comment]
PTAL

File: opencensus/trace/internal/trace_context_test.cc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  EXPECT_THAT(ctx, IsValid());
+  EXPECT_EQ("404142434445464748494a4b4c4d4e4f-6162636465666768-00",
+            ctx.ToString());
+  EXPECT_EQ(header, ToTraceParentHeader(ctx));
+}
+
+TEST(TraceParentTest, ExpectedFailures) {
+#define INVALID(str) EXPECT_THAT(FromTraceParentHeader(str), IsInvalid())
+  INVALID("");
+  INVALID("AB-404142434445464748494A4B4C4D4E4F-6162636465666768-00")
```
Comment (bogdandrutu) [inline opencensus/trace/internal/trace_context_test.cc]: Looks like here you used uppercase. I filed an issue to determine exactly the right thing here:
https://github.com/w3c/trace-context/issues/219

Comment (g-easy) [reply opencensus/trace/internal/trace_context_test.cc]: The parser handles both and the serializer produces lowercase. Should this PR wait for resolution of that issue?

Comment (bogdandrutu) [reply opencensus/trace/internal/trace_context_test.cc]: I think the issue was fixed. Accept lowercase preferable.

Comment (g-easy) [reply opencensus/trace/internal/trace_context_test.cc]: I've changed the parser to only accept lowercase, per the trace-context spec.

**Pull request merged**